src/components/analytics/AnalyticsCharts.jsx

```diff
@@ -11,33 +10,51 @@ import AnalyticsApiService from './data/service';
 
 // eslint-disable-next-line no-unused-vars
 export default function AnalyticsCharts(enterpriseId) {
-  const [token, setToken] = useState(null);
+  const [tokenUsedOnce, setTokenUsedOnce] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState(null);
-  // Fetch token
+  const ref = useRef(document.getElementById('tableau_frame'));
```

Comment:
I think this is not the right way of using this hook with `document.getElementById`. We can do it like this

```
  const tableauRef = useRef(null);
  <div id="tableau_frame" ref={tableauRef} />
```

Reply:
done